packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -101,12 +100,20 @@ async function getFileLinkCore(
             let additionalProps;
             const fileLink = await getWithRetryForTokenRefresh(async (options) => {
                 attempts++;
-                const token = await getToken({ ...options, ... odspUrlParts });
+                const storageTokenFetcher = toInstrumentedOdspTokenFetcher(
+                    logger,
+                    odspUrlParts,
+                    getToken,
+                    true /* throwOnNullToken */,
+                );
+                const storageToken = await storageTokenFetcher(options,"GetSharingLinkTokenCore");
+                assert(storageToken !== null, "Storage token should not be null");
```

Comment:
We reserve assert for validating internal invariants that should be true no matter what the input (e.g. valid state machine transitions).  In this case, we need to raise a proper error object with an `errorType`, since this depends on the callback passed to us.

...Ok looking closer, this is ok as an assert, because the instrumented fetcher should never return null - it will throw the proper error.  Let's update the assert message then.

```suggestion
                assert(storageToken !== null, "Instrumented token fetcher with throwOnNullToken = true should never return null");
```